### PR TITLE
refactor(#2246): Refactor InfluxStore.onEvent

### PR DESCRIPTION
**Pull request by tenthe**
### Purpose

Adds unit tests for the class `InfluxStore` and refactored the code.

### Remarks

PR introduces (a) breaking change(s): yes

PR introduces (a) deprecation(s): no

### CODE REVIEW

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java [hunk 1]
```diff
@@ -50,20 +50,32 @@ public class InfluxStore {
 
   private RawFieldSerializer rawFieldSerializer = new RawFieldSerializer();
 
-  public InfluxStore(DataLakeMeasure measure,
-                     InfluxConnectionSettings settings) {
+  /**
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java:53]: Can we get rid of this testing-only constructor?
E.g., by using Iflux connection provider?

Comment (tenthe) [reply streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java:53]: Ok, thanks for the feedback and the suggestion.
I will try my best ;)

File: streampipes-extensions/streampipes-connectors-influx/src/main/java/org/apache/streampipes/extensions/connectors/influx/sink/InfluxDbClient.java [hunk 2]
```diff
@@ -67,7 +70,7 @@ private void connect() throws SpRuntimeException {
     var databaseName = connectionSettings.getDatabaseName();
 
     // Checking whether the database exists
-    if (!InfluxRequests.databaseExists(influxDb, databaseName)) {
+    if (!influxClientProvider.databaseExists(influxDb, databaseName)) {
       LOG.info("Database '" + databaseName + "' not found. Gets created ...");
       createDatabase(databaseName);
```
Comment (bossenti) [inline streampipes-extensions/streampipes-connectors-influx/src/main/java/org/apache/streampipes/extensions/connectors/influx/sink/InfluxDbClient.java:75]: Can we use `createDatabase()` from the InfluxClientProvider here?

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxClientProvider.java [hunk 3]
```diff
@@ -58,7 +67,80 @@ public static InfluxDB getInfluxDBClient(InfluxConnectionSettings settings) {
...
+      }
+    }
+    return false;
+  }
+
+
   private static Environment getEnvironment() {
     return Environments.getEnvironment();
   }
+
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxClientProvider.java:144]:
```suggestion
```

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxClientProvider.java [hunk 4]
```diff
@@ -58,7 +67,80 @@ public static InfluxDB getInfluxDBClient(InfluxConnectionSettings settings) {
...
+
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxClientProvider.java:145]:
```suggestion
```

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java [hunk 5]
```diff
@@ -20,219 +20,161 @@
...
 public class InfluxStore {
 
   private static final Logger LOG = LoggerFactory.getLogger(InfluxStore.class);
-  DataLakeMeasure measure;
-  Map<String, String> sanitizedRuntimeNames = new HashMap<>();
+
+  private DataLakeMeasure measure;
+  private List<EventProperty> allEventProperties;
+  private Map<String, String> sanitizedRuntimeNames = new HashMap<>();
   private InfluxDB influxDb = null;
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java:46]:
```suggestion
  private final InfluxDB influxDb;
```

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java [hunk 6]
```diff
@@ -20,219 +20,161 @@
...
-                     Environment environment) throws SpRuntimeException {
-    this(measure, InfluxConnectionSettings.from(environment));
+  public InfluxStore(
+      DataLakeMeasure measure,
+      Environment environment,
+      InfluxClientProvider influxClientProvider
+  ) throws SpRuntimeException {
+    this.measure = measure;
+    storeSanitizedRuntimeNames(measure);
+    this.allEventProperties = getAllEventPropertiesExceptTimestamp(measure);
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java:58]:
```suggestion
   allEventProperties = getAllEventPropertiesExceptTimestamp(measure);
```

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java [hunk 7]
```diff
@@ -20,219 +20,161 @@
...
-   * @param dbName The name of the database which should be created
+   * Logs all fields which are present in the schema, but not in the provided event
    */
-  private void createDatabase(String dbName) throws SpRuntimeException {
-    if (!dbName.matches("^[a-zA-Z_]\\w*$")) {
-      throw new SpRuntimeException(
-          "Database name '" + dbName + "' not allowed. Allowed names: ^[a-zA-Z_][a-zA-Z0-9_]*$");
+  private void logMissingFields(Event event) {
+    var missingFields = getMissingProperties(allEventProperties, event);
+    if (missingFields.size() > 0) {
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java:95]:
```suggestion
    if (!missingFields.isEmpty()) {
```

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java [hunk 8]
```diff
@@ -20,219 +20,161 @@
...
+        })
+        .map(EventProperty::getRuntimeName)
+        .collect(Collectors.toList());
 
-    // sanitize event
-    for (var key : event.getRaw().keySet()) {
-      if (InfluxDbReservedKeywords.KEYWORD_LIST.stream().anyMatch(k -> k.equalsIgnoreCase(key))) {
-        event.renameFieldByRuntimeName(key, key + "_");
-      }
+    if (nullFields.size() > 0) {
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java:123]:
```suggestion
    if (!nullFields.isEmpty()) {
```

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java [hunk 9]
```diff
@@ -20,219 +20,161 @@
...
+  /**
+   * Returns a list of the runtime names that are missing within the event
+   */
+  private List<String> getMissingProperties(
+      List<EventProperty> allEventProperties,
+      Event event
+  ) {
+    return allEventProperties.stream()
+                             .map(EventProperty::getRuntimeName)
+                             .filter(runtimeName -> !event.getOptionalFieldByRuntimeName(runtimeName)
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java:167]:
```suggestion
                             .filter(runtimeName -> event.getOptionalFieldByRuntimeName(runtimeName)
```

File: streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java [hunk 10]
```diff
@@ -20,219 +20,161 @@
...
+                                                          .isPresent())
```
Comment (bossenti) [inline streampipes-data-explorer-commons/src/main/java/org/apache/streampipes/dataexplorer/commons/influx/InfluxStore.java:168]:
```suggestion
                                                          .isEmpty())
```

**Review by bossenti — CHANGES_REQUESTED** [review]
Thank you very much for this refactoring @tenthe!
This improves the code here a lot 

I've left some minor remarks and ideas for improvements 🙂

File: streampipes-extensions/streampipes-connectors-influx/src/main/java/org/apache/streampipes/extensions/connectors/influx/sink/InfluxDbClient.java (hunk 2 above)
Comment (tenthe) [reply streampipes-extensions/streampipes-connectors-influx/src/main/java/org/apache/streampipes/extensions/connectors/influx/sink/InfluxDbClient.java:75]: No, this only makes a check if the database does exist

**Review by dominikriemer — APPROVED** [review]

**Pull request merged**